### PR TITLE
chore(optimize8): fix optimize java checks

### PR DESCRIPTION
**Pull request by buccarel**
## Description

This fixes the following workflow:
* `Optimize Run Java checks / Run Java checks (pull_request)`

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -57,6 +57,7 @@
               <exclude>clients/go/vendor/**/*.md</exclude>
               <exclude>**/node_modules/**/*.md</exclude>
               <exclude>tasklist/client/**/*.md</exclude>
+              <exclude>**/.node/**/*.md</exclude>
```
Comment (PHWaechtler) [inline pom.xml:60]: Is this also related to the java checks?

File: .github/workflows/optimize-run-java-checks.yml [hunk 2]
```diff
@@ -31,7 +31,7 @@ jobs:
         uses: actions/setup-java@99b8673ff64fbf99d8d325f52d9a5bdedb8483e9 # v4
         with:
           distribution: "adopt"
-          java-version: "17"
+          java-version: "21"
```
Comment (PHWaechtler) [inline .github/workflows/optimize-run-java-checks.yml:34]: Does this mean we're changing our java version? Would that need documented?
https://docs.camunda.io/docs/reference/supported-environments/#component-requirements

**Review by HeleneW-dot — CHANGES_REQUESTED** [review]
Job lgtm, but I have some changes as its not clear to me what it means to change the version there. If it only affects Ci then all good to merge

File: .github/workflows/optimize-run-java-checks.yml (hunk 2 above)
Comment (buccarel) [reply .github/workflows/optimize-run-java-checks.yml:34]: I thought I did it here time ago with this:
https://github.com/camunda/camunda-docs/commit/f56d15e7ef9d57853f0928bd456c82818d9031ac

I don't know why the web site is not

Comment (PHWaechtler) [reply .github/workflows/optimize-run-java-checks.yml:34]: ok all good then, just please make sure its updated

**Review by HeleneW-dot — APPROVED** [review]
